Calibration/HcalCalibAlgos/plugins/HcalIsoTrackAnalyzer.cc

```diff
@@ -0,0 +1,308 @@
+// system include files
+#include <atomic>
+#include <cmath>
+#include <memory>
+#include <string>
+#include <vector>
+
+// Root objects
+#include "TROOT.h"
+#include "TSystem.h"
+#include "TFile.h"
+#include "TProfile.h"
+#include "TDirectory.h"
+#include "TTree.h"
+#include "TLorentzVector.h"
+#include "TInterpreter.h"
+
+#include "DataFormats/HcalCalibObjects/interface/HcalIsoTrkCalibVariables.h"
+#include "DataFormats/HcalCalibObjects/interface/HcalIsoTrkEventVariables.h"
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/one/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+
+//#define EDM_ML_DEBUG
+
+class HcalIsoTrackAnalyzer : public edm::one::EDAnalyzer<edm::one::WatchRuns, edm::one::SharedResources> {
+public:
+  explicit HcalIsoTrackAnalyzer(edm::ParameterSet const&);
+  ~HcalIsoTrackAnalyzer() override {}
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+private:
+  void analyze(edm::Event const&, edm::EventSetup const&) override;
+  void beginJob() override;
+  void beginRun(edm::Run const&, edm::EventSetup const&) override {}
+  void endRun(edm::Run const&, edm::EventSetup const&) override;
+
+  edm::Service<TFileService> fs;
+  const double pTrackLow_, pTrackHigh_;
+  const int useRaw_, dataType_;
+  const edm::InputTag labelIsoTkVar_, labelIsoTkEvt_;
+  edm::EDGetTokenT<HcalIsoTrkCalibVariablesCollection> tokIsoTrkVar_;
+  edm::EDGetTokenT<HcalIsoTrkEventVariablesCollection> tokIsoTrkEvt_;
+  unsigned int nRun_, nRange_, nLow_, nHigh_;
+
+  TTree *tree, *tree2;
+  int t_Run, t_Event, t_DataType, t_ieta, t_iphi;
+  int t_goodPV, t_nVtx, t_nTrk;
+  double t_EventWeight, t_p, t_pt, t_phi;
+  double t_l1pt, t_l1eta, t_l1phi;
+  double t_l3pt, t_l3eta, t_l3phi;
+  double t_mindR1, t_mindR2;
+  double t_eMipDR, t_eMipDR2, t_eMipDR3, t_eMipDR4;
+  double t_eMipDR5, t_hmaxNearP, t_gentrackP;
+  double t_emaxNearP, t_eAnnular, t_hAnnular;
+  double t_eHcal, t_eHcal10, t_eHcal30, t_rhoh;
+  bool t_selectTk, t_qltyFlag, t_qltyMissFlag, t_qltyPVFlag;
+  const std::vector<unsigned int>*t_DetIds, *t_DetIds1, *t_DetIds3;
+  const std::vector<double>*t_HitEnergies, *t_HitEnergies1, *t_HitEnergies3;
+  const std::vector<bool>* t_trgbits;
+
+  unsigned int t_RunNo, t_EventNo;
+  bool t_TrigPass, t_TrigPassSel, t_L1Bit;
+  int t_Tracks, t_TracksProp, t_TracksSaved;
+  int t_TracksLoose, t_TracksTight, t_allvertex;
+  const std::vector<int>*t_ietaAll, *t_ietaGood, *t_trackType;
+  const std::vector<bool>* t_hltbits;
```

Comment:
@bsunanda I think these member variable should end with an _. Please see coding rule 2.9
"A trailing “_” is the preferred method to distinguish a data member from the getter method"

Reply:
We have many macros which work with this naming convention (are used for past several years). Cannot change the names now

Reply:
Not changing that for backward compatibility of many macros and used by other tree analyzers